tests/DrupalConsoleStackTest.php

```diff
@@ -0,0 +1,405 @@
+    use CommandArguments;
+
+    /**
+     * Set up the Robo container so that we can create tasks in our tests.
+     */
+    public function setUp()
+    {
+        $container = Robo::createDefaultContainer(null, new NullOutput());
+        $this->setContainer($container);
+    }
+
+    /**
+     * Scaffold the collection builder.
+     *
+     * @return \Robo\Collection\CollectionBuilder
+     *   The collection builder.
+     */
+    public function collectionBuilder()
+    {
+        $emptyRobofile = new \Robo\Tasks;
+        return $this->getContainer()
+            ->get('collectionBuilder', [$emptyRobofile]);
+    }
+
+    /**
+     * Tests a given command.
+     *
+     * @param string $method
+     *   The method to call on DrupalConsoleStack.
+     * @param string $command
+     *   The expected command, without the 'drupal ' prefix and ' --yes' suffix.
+     * @param array $args
+     */
+    protected function doTestCommand($method, $command, array $args = [])
+    {
+        $stack = $this->taskDrupalConsoleStack();
+        call_user_func_array([$stack, $method], $args);
+        $expected = 'drupal ' . $command . ' --yes';
+        $this->assertEquals($expected, $stack->getCommand());
+    }
+
+    /**
+     * Test a given option.
+     *
+     * @param string $method
+     *   The method to call on DrupalConsoleStack.
+     * @param string $option
+     *   The option name.
+     * @param string $value
+     *   The option value.
+     */
+    protected function doTestOption($method, $option, $value = null)
+    {
+        $stack = $this->taskDrupalConsoleStack();
+        call_user_func_array([$stack, $method], [$value]);
+        $expected = 'drupal command --' . $option;
+        if (!is_null($value)) {
+            $expected .= ' ' . static::escape($value);
+        }
+        $stack->drupal('command', false);
+        $this->assertEquals($expected, $stack->getCommand());
+    }
+
+    /**
+     * Test if the yes option is assumed by default.
+     */
+    public function testYesIsAssumed()
+    {
+        $command = $this->taskDrupalConsoleStack()
+          ->drupal('command')
+          ->getCommand();
+        $this->assertEquals('drupal command --yes', $command);
+    }
+
+    /**
+     * Test the absence of the yes option if explicitly declared.
+     */
+    public function testAbsenceofYes()
+    {
+        $command = $this->taskDrupalConsoleStack()
+          ->drupal('command', false)
+          ->getCommand();
+        $this->assertEquals('drupal command', $command);
+    }
+
+    /**
+     * Test if options are prepended before each command.
+     */
+    public function testOptionsArePrependedBeforeEachCommand()
+    {
+        $command = $this->taskDrupalConsoleStack()
+          ->drupalRootDirectory('/var/www/html/app')
+          ->drupal('command-1')
+          ->drupal('command-2')
+          ->getCommand();
+        $this->assertEquals(2, preg_match_all('#--root /var/www/html/app#', $command));
```

Comment:
use `$this->assertRegExp() `

See https://phpunit.de/manual/current/en/appendixes.assertions.html